### ISSUE TITLE
jasmine.objectContaining no longer works with inherited properties

### ISSUE THREAD

**Issue by Lalem001** [jasmine/jasmine#787]
I have many tests that have suddenly started failing in Jasmine 2.2.0. I have written a simple example case below.

Given:
```js
function SomeClass () {}
someClass.prototype.someProp = 'foo'
```

Test:
```js
var someClass = new SomeClass();
expect(someClass.someProp).toEqual('foo'); // pass
expect(someClass).toEqual(jasmine.objectContaining({ someProp: 'foo' })); // fail
```

I know why it is failing, since the source changed the comparison conditions from [`hasKey`](https://github.com/jasmine/jasmine/blob/b56c2a69ac8ca6dab78418e71bfbdd7ee79f5336/src/core/ObjectContaining.js#L13) to the native [`hasOwnProperty`](https://github.com/jasmine/jasmine/blob/8c59875afe3968a61ddcf445762a90de53cf8951/src/core/asymmetric_equality/ObjectContaining.js#L24).

Was this change intentional? If so, is it documented anywhere?

**Comment by Gerg** [comment]
I believe this was fixed by 69956bf8f61a0277ef3134126d36ece8fc9dde88.

**Comment by slackersoft** [comment]
We think this should be fixed in master now, so I'm going to close this. If you have a chance, please try this again on master. If it still doesn't work, we'd like to know, as this regression was not intentional.

Closing for now.

**Comment by Lalem001** [comment]
It was indeed fixed by 69956bf8f61a0277ef3134126d36ece8fc9dde88. Sorry for the delayed response.
